root/static/scripts/edit/utility/subfieldErrors.js

```diff
@@ -7,29 +7,79 @@
  * later version: http://www.gnu.org/licenses/gpl-2.0.txt
  */
 
-import {
-  type FormOrAnyFieldT,
-  type WritableAnyFieldT,
-  type WritableFormOrAnyFieldT,
-  iterSubfields,
-  iterWritableSubfields,
-} from './iterSubfields.js';
+import type {CowContext} from 'mutate-cow';
+
+function* iterSubfields(
+  formOrField: FormOrAnyFieldT,
+): Generator<AnyFieldT, void, void> {
+  switch (formOrField.type) {
+    case 'compound_field':
+      yield formOrField;
+      // falls through
+    case 'form':
+      for (const subfield of Object.values(formOrField.field)) {
+        yield* iterSubfields(subfield);
+      }
+      break;
+    case 'field':
+      yield formOrField;
+      break;
+    case 'repeatable_field': {
+      yield formOrField;
+      for (const subfield of formOrField.field) {
+        yield* iterSubfields(subfield);
+      }
+      break;
+    }
+  }
+}
+
+function* iterContextSubfields(
+  formOrFieldCtx: CowContext<FormOrAnyFieldT>,
+): Generator<CowContext<AnyFieldT>, void, void> {
+  const formOrField = formOrFieldCtx.read();
+  switch (formOrField.type) {
+    case 'compound_field':
+      // $FlowIgnore[incompatible-type-arg]
```

Comment:
Basically, checking `formOrField.type` doesn't refine `formOrFieldCtx`'s type parameter, so Flow thinks it can still refer to, e.g., a form rather than a compound field.